src/arithmetic.ts

```diff
@@ -0,0 +1,74 @@
+export interface Arithmetic<NumberType> {
```

Comment:
Mind adding a comment that describes what this interface is about to make it easier for future readers to understand?

Reply:
I will add some comments in the coming week.
But there was something I didn't like about this interface, and I finally see how to fix it.

The core part of the application `unit.ts` does not need at all the geometric part.
So if someone want to use and create his custom types, he needs to define the geometric part, but there is no need. The geometric part is only needed by the "angle" is to will be a second parameter to his `withValueType`.

So I will split this interface in two: "Arithmetic" and "Geometric".